presto-main/src/main/java/io/prestosql/server/security/CertificateAuthenticatorManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logger;
+import io.prestosql.spi.security.CertificateAuthenticator;
+import io.prestosql.spi.security.CertificateAuthenticatorFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static io.airlift.configuration.ConfigurationLoader.loadPropertiesFrom;
+import static java.util.Objects.requireNonNull;
+
+public class CertificateAuthenticatorManager
+{
+    private static final Logger log = Logger.get(CertificateAuthenticatorManager.class);
+
+    private static final File CONFIG_FILE = new File("etc/certificate-authenticator.properties");
+    private static final String NAME_PROPERTY = "certificate-authenticator.name";
+
+    private final AtomicBoolean required = new AtomicBoolean();
+    private final Map<String, CertificateAuthenticatorFactory> factories = new ConcurrentHashMap<>();
+    private final AtomicReference<CertificateAuthenticator> authenticator = new AtomicReference<>();
+
+    public void setRequired()
+    {
+        required.set(true);
+    }
+
+    public void addCertificateAuthenticatorFactory(CertificateAuthenticatorFactory factory)
+    {
+        checkArgument(factories.putIfAbsent(factory.getName(), factory) == null,
+                "Certificate authenticator '%s' is already registered", factory.getName());
+    }
+
+    public void loadCertificateAuthenticator()
+            throws Exception
+    {
+        if (!required.get()) {
+            return;
+        }
+
+        File configFile = CONFIG_FILE.getAbsoluteFile();
+        Map<String, String> properties = new HashMap<>(loadPropertiesFrom(configFile.getPath()));
```

Comment:
I talked to @dain and we agreed that a custom certificate authenticator is an advanced feature and we shouldn't require explicitly configuring the default one. Relying on file existence is no different than we already for other features like event listeners, catalogs, etc.